Assembler.Base/BaseMessageHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Assembler.Core;
+using Assembler.Core.Entities;
+
+namespace Assembler.Base
+{
+    public abstract class BaseMessageHandler<TFrame, TMessage> : IHandler<TFrame, TMessage>
+        where TFrame : BaseFrame
+        where TMessage : BaseMessageInAssembly
+    {
+        protected readonly ITimeBasedCache<TMessage> Cache;
+        protected readonly IFactory<TFrame, string> IdentifierFactory;
+        protected readonly IMessageEnricher<TFrame, TMessage> MessageEnricher;
+        protected readonly ICreator<TMessage> AssembledMessageCreator;
+
+        public event Action<TMessage> MessageAssemblyFinished;
+
+        protected BaseMessageHandler(ITimeBasedCache<TMessage> cache, IFactory<TFrame, string> identifierFactory,
+            IMessageEnricher<TFrame, TMessage> messageEnricher, ICreator<TMessage> assembledMessageCreator)
+        {
+            Cache = cache;
+            IdentifierFactory = identifierFactory;
+            MessageEnricher = messageEnricher;
+            AssembledMessageCreator = assembledMessageCreator;
+        }
+
+        protected void ReleaseMessage(TMessage message)
+        {
+            MessageAssemblyFinished?.Invoke(message);
+        }
+
+        protected string GetIdentifier(TFrame frame) => IdentifierFactory.Create(frame);
+
+        protected TMessage CreateMessage() => AssembledMessageCreator.Create();
+
+        public abstract void Handle(TFrame frame);
```

Comment:
Although this is abstract, I usually prefer that public methods would appear before protected methods

Reply:
This looks weird, but I changed it.

Reply:
Now it's in-between 2 protected methods lol

Reply:
f

Reply:
I meant it would be after the c'tor and before the protected methods

מצטער ששיגעתי אותך

Reply:
לא לא, זה טוב!